wallet/manager.go

```diff
@@ -327,33 +327,37 @@ func syncStore(ctx context.Context, store Store, cm ChainManager, index types.Ch
 }
 
 // NewManager creates a new wallet manager.
-func NewManager(cm ChainManager, store Store, opts ...Option) (*Manager, error) {
+func NewManager(opts ...Option) (*Manager, error) {
 	m := &Manager{
 		indexMode:     IndexModePersonal,
 		syncBatchSize: defaultSyncBatchSize,
 
-		chain: cm,
-		store: store,
-		log:   zap.NewNop(),
-		tg:    threadgroup.New(),
+		log: zap.NewNop(),
+		tg:  threadgroup.New(),
 	}
 
 	for _, opt := range opts {
 		opt(m)
 	}
 
+	if m.chain == nil {
+		panic("chain manager is required")
+	} else if m.store == nil {
+		panic("store is required")
+	}
```

Comment:
I don't think these need to be Options. Functions can have both mandatory and optional parameters.

Reply:
Yeah, option-ifying all the things was a not so nice workaround to dealing with both the remote and local codepaths in the node startup. 😅 

It felt less awful than a local embedded interface
```go
var cm interface {
	wallet.ChainManager
	api.ChainManager
}
```